pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -138,15 +151,55 @@ func (pfactory *PredicateMetadataFactory) GetMetadata(pod *v1.Pod, nodeNameToInf
 	return predicateMetadata
 }
 
+func (topologyPairsMaps *topologyPairsMaps) AddTopologyPair(pair topologyPair, pod *v1.Pod) {
+	found := false
+	for _, existingPod := range topologyPairsMaps.topologyPairToPods[pair] {
+		if existingPod == pod {
+			found = true
+			break
+		}
+	}
+	if !found {
+		topologyPairsMaps.topologyPairToPods[pair] = append(topologyPairsMaps.topologyPairToPods[pair], pod)
+		topologyPairsMaps.podToTopologyPairs[schedutil.GetPodFullName(pod)] = append(topologyPairsMaps.podToTopologyPairs[schedutil.GetPodFullName(pod)], pair)
```

Comment:
Definitely using a set would provide easier and faster look-ups in this case as mentioned. However, If the pod is not found in the first map, it should not be in the second map as well since both maps are kept in sync (additions/deletions) so we do not need to iterate over the second map. Also, a pod object is needed to get its anti-affinity terms, so using a string as the key for podSet might not work.